### PR TITLE
Recorder: 1 second period with 1 decimal place

### PR DESCRIPTION
**Pull request by thyttan**
This will make logs with 1 second periods record time with 1 decimal place precision.

EDIT3 - my first solution seemed to behave weirdly to me. But after sitting on it I realized it was probably just floating point number math doing what it does. The discussion that followed here no longer applies.

### CODE REVIEW

**Comment by thyttan** [comment]
So I guess it was just floating point math doing its thing, and not wrong per se. To get nicer formatting in the csv file, I've converted to string. 

Seems to give good results now.

**Comment by bobrippling** [comment]
Yeah that's floating point - there's no way to represent some numbers with that level of precision, and the nearest representable number might not be displayable in decimal (like how we can't write 1/3 in decimal)

So:
```javascript
Math.round(n * 10) / 10
```

won't necessarily give you a number with a single decimal place. If you want that, you could try [`n.toFixed(1)`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toFixed), which will give you exactly 1 decimal place (in a string), for popping into the CSV if you prefer

File: apps/recorder/widget.js [hunk 1]
```diff
@@ -176,11 +176,17 @@
...
 
-  let writeLog = function() {
+  let writeLog = function(period) {
     entriesWritten++;
     WIDGETS["recorder"].draw();
     try {
-      var fields = [Math.round(getTime())];
+      var time = getTime();
+      if (period===1) {
+        time = Math.round(time)+"."+(Math.round(time*10))%10;
```
Comment (thyttan) [inline apps/recorder/widget.js:185]: @bobrippling do you think it's better to use [n.toFixed(1)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toFixed) here like you hinted?

Comment (bobrippling) [reply apps/recorder/widget.js:185]: Yeah I could see that being more consistent for giving a single decimal place and I believe it'll produce the same answer we're after

**Comment by thyttan** [comment]
I tested the current code by recoring just the time and battery status with periods 1 second and 2 seconds for ca 10 seconds each. Seems to work as it should.

**Comment by bobrippling** [comment]
Nice - those changes LGTM, shall see if Gordon has anything to mention though  👀

**Comment by bobrippling** [comment]
I think let's merge - we can pick up other parts in #2819 if needed

**Pull request merged**